src/main/java/software/amazon/smithy/lsp/Main.java

```diff
@@ -61,21 +65,28 @@ public static void main(String[] args) {
         Socket socket = null;
         InputStream in;
         OutputStream out;
-
+        List<String> argList = Arrays.asList(args);
         try {
-            String port = args[0];
+            Optional<Exception> launchFailure;
+            String port = getOrDefault(argList, 0, "0");
+            String type = getOrDefault(argList, 1, null);
```

Comment:
If the websocket path returns right away, maybe we can hoist that out of the if-else into a guard clause, and not change the existing if-else.
```suggestion
            String port = getOrDefault(argList, 0, "0");
            String type = getOrDefault(argList, 1, null);
            
            if ("--ws".equals(type)) {
              // Do websocket stuff
              return;
            }
            
            // Rest is unchanged
```